plugin/storage/es/spanstore/writer.go

```diff
@@ -76,6 +77,7 @@ type Span struct {
 // NewSpanWriter creates a new SpanWriter for use
 func NewSpanWriter(
 	client es.Client,
+	dateFormat string,
```

Comment:
since we're breaking the API, let's use a struct instead
```
type SpanWriterOptions struct {
 	logger *zap.Logger
 	metricsFactory metrics.Factory
 	dateFormat string
 	numShards int64
 	numReplicas int64
}

func NewSpanWriter(client es.Client, options SpanWriterOptions) {}
```

It will be easier to extend if we want to make index name configurable